test/logger.js

```diff
@@ -84,6 +84,51 @@ test("A logger instance shouldn't log if namespace is not enabled", t => {
     logger.internals.write.restore()
 })
 
+test("A logger instance should't log if log level is lower than namespace pattern level", t => {
+    logger.setNamespaces('error@test:*')
+
+    logger.setLevel('info')
+
+    const log = logger('test:subtest')
+    const spy = sinon.spy(logger.internals, 'write')
+
+    log.info(null, 'test')
+
+    t.is(spy.callCount, 0)
+
+    logger.internals.write.restore()
+})
+
+test("A logger instance should log if log level is higher than namespace pattern level", t => {
+    logger.setNamespaces('debug@test:*')
+
+    logger.setLevel('info')
+
+    const log = logger('test:subtest')
+    const spy = sinon.spy(logger.internals, 'write')
+
+    log.debug(null)
+    t.true(spy.calledOnce)
+
+    logger.internals.write.restore()
+})
+
+    test("A logger instance should log according to state defined in the latest matching namespace in the list", t => {
+    logger.setNamespaces('warn@test:*,test2:*,error@test:*,-test2:*')
```

Comment:
It's just a proposal, if you're more comfortable with the micro DSL, it's fine :)